### PR TITLE
cclient: Use flex_hash_to_tryte to transfer flex to trytes

### PR DESCRIPTION
**Pull request by jkrvivian**
There's a `flex_hash_to_trytes` can be used in transfering flex_hash to trytes. 

# Test Plan:
Pass `cclient/serialization` test.

### CODE REVIEW

File: cclient/serialization/json/json_serializer.c [hunk 1]
```diff
@@ -106,12 +107,10 @@ static retcode_t flex_hash_array_to_json_array(flex_hash_array_t* const head,
...
       size_t len_trytes = elt->hash->num_trits / 3;
-      trit_t trytes_out[len_trytes + 1];
-      size_t trits_count =
-          flex_trits_to_trytes(trytes_out, len_trytes, elt->hash->trits,
-                               elt->hash->num_trits, elt->hash->num_trits);
+      char trytes_out[len_trytes + 1];
+      ret = flex_hash_to_trytes(elt->hash, trytes_out);
       trytes_out[len_trytes] = '\0';
-      if (trits_count != 0) {
+      if (ret == RC_OK) {
```
Comment (th0br0) [inline cclient/serialization/json/json_serializer.c:119]: Could you adjust the error code below `RC_CCLIENT_FLEX_TRITS` and give it a better name as well?

**Review by th0br0 — CHANGES_REQUESTED** [review]

**Comment by th0br0** [comment]
Hey @jkrvivian - thanks for the PR! Looks like a very useful change :)

**Comment by jkrvivian** [comment]
Sure!

File: cclient/serialization/json/json_serializer.c (hunk 1 above)
Comment (jkrvivian) [reply cclient/serialization/json/json_serializer.c:119]: @th0br0 ,
Now, `RC_CCLIENT_FLEX_TRITS` error is returned when trits num is invalid.
How about changing it to `RC_CCLIENT_INVALID_TRITS_LEN`?

And in function `flex_hash_to_trytes`, we will first check if `char* trytes` is null or not, now it returns `RC_CCLIENT_FLEX_TRITS`. But I think `RC_CCLIENT_NULL_PTR` will be more appropriate.

**Review by th0br0 — DISMISSED** [review]

**Comment by thibault-martinez** [comment]
Hi @jkrvivian , thanks for the PR ! Could you fix the coding style error ?

**Review by th0br0 — DISMISSED** [review]

**Comment by th0br0** [comment]
Sorry for taking so long to merge this! Thanks a lot for your contribution.

**Review by th0br0 — APPROVED** [review]

**Comment by th0br0** [comment]
This doesn't build? Please fix!

**Comment by oopsmonk** [comment]
@jkrvivian thanks for yur PR, but flex_hash is deprecated in #473 and it will be removed in a cleanup on cclient.

**Pull request closed without merging**